src/field_theory/krull_topology.lean

```diff
@@ -287,3 +286,60 @@ begin
 end
 
 end totally_disconnected
+
+
+section finite
+namespace krull_topology
+
+variables {K E L : Type*} [field K] [field E] [algebra K E] [field L] [algebra K L]
+
+/-- A technical finiteness result. -/
+noncomputable def fintype.subtype_prod {E : Type*} {X : set E} (hX : X.finite) {L : Type*}
+  (F : E → multiset L) : fintype (Π x : X, {l : L // l ∈ F x}) :=
+by { classical, letI : fintype X := set.finite.fintype hX, exact pi.fintype}
+
+variable (K)
+
+variable [finite_dimensional K E]
+
+variables (E L)
+
+/-- Function from Hom_K(E,L) to pi type Π (x : basis), roots of min poly of x -/
+def roots_of_min_poly_pi_type (φ : E →ₐ[K] L)
+(x : set.range (finite_dimensional.fin_basis K E : _ → E)) :
+  {l : L // l ∈ (((minpoly K x.1).map (algebra_map K L)).roots : multiset L)} :=
+⟨φ x, begin
+  rw [polynomial.mem_roots_map (minpoly.ne_zero_of_finite_field_extension K x.val),
+    ← polynomial.alg_hom_eval₂_algebra_map, ← φ.map_zero],
+  exact congr_arg φ (minpoly.aeval K (x : E)),
+end⟩
+
+lemma aux_inj_roots_of_min_poly : function.injective (roots_of_min_poly_pi_type K E L) :=
+begin
+  intros f g h,
+  suffices : (f : E →ₗ[K] L) = g,
+  { rw linear_map.ext_iff at this,
+    ext x, exact this x },
+  rw function.funext_iff at h,
+  apply linear_map.ext_on (finite_dimensional.fin_basis K E).span_eq,
+  rintro e he,
+  have := (h ⟨e, he⟩),
+  apply_fun subtype.val at this,
+  exact this,
+end
+
+/-- Given field extensions `E/K` and `L/K`, with `E/K` finite, there are finitely many `K`-algebra
+  homomorphisms `E →ₐ[K] L`. -/
+noncomputable instance fintype_alg_hom : fintype (E →ₐ[K] L) :=
```

Comment:
I think this is already somewhere, maybe with additional assumptions, can you check it? In any case it deserves to be in a more fundamental file.

Reply:
It's [field.alg_hom.fintype](https://leanprover-community.github.io/mathlib_docs/field_theory/primitive_element.html#field.alg_hom.fintype). Your result is strictly more general, you can just generalize it.

Reply:
This is probably because I really [like](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/An.20example.20of.20why.20formalization.20is.20useful/near/277293484) the generalization and Lean hates me.

But let me check.